master/internal/core.go

```diff
@@ -929,9 +930,18 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 		SegmentEnabled:        m.config.Telemetry.Enabled && m.config.Telemetry.SegmentMasterKey != "",
 		SegmentAPIKey:         m.config.Telemetry.SegmentMasterKey,
 	}
+	if m.config.Logging.Retention != nil {
+		m.taskSpec.LogRetention = m.config.Logging.Retention.Duration
+	}
```

Comment:
why not set these under the same conditional? if there is some reason, we should document it.